nexus/types/src/deployment/clickhouse.rs

```diff
@@ -0,0 +1,118 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Types used in blueprints related to clickhouse configuration
+
+use clickhouse_admin_types::{KeeperId, ServerId};
+use omicron_common::api::external::Generation;
+use omicron_uuid_kinds::OmicronZoneUuid;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+use uuid::Uuid;
+
+/// Global configuration for all clickhouse servers (replicas) and keepers
+#[derive(Clone, Debug, Eq, PartialEq, JsonSchema, Deserialize, Serialize)]
+pub struct ClickhouseClusterConfig {
+    /// The last update to the clickhouse cluster configuration
+    ///
+    /// This is used by `clickhouse-admin` in the clickhouse server and keeper
+    /// zones to discard old configurations.
+    pub generation: Generation,
+    /// Clickhouse Server ids must be unique and are handed out monotonically.
```

Comment:
Nit

```suggestion
    /// Clickhouse Server IDs must be unique and are handed out monotonically.
```

Reply:
Fixed in [340046e](https://github.com/oxidecomputer/omicron/pull/6566/commits/340046eac678f712d813de8a6bcb7ed8fbb5435a)